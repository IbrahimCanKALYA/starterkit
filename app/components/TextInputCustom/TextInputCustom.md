# TextInputCustom

## Description

- Simple textInput component allows you changing border color onFocus & onBlur.

## Functions

- onChangeText() : Will be trigerred whenever text changes.

- getText() : Will return textInput value as text.

- setText() : Sets text of component.

- focus() : Focuses on the textInput.

- clear() : Clears the textInput value.

## Props

- onChangeText(Function) : Will be trigerred inside of the onChangeText() function whenever text changes.
- placeholder(String) : Placeholder of the textInput
- secureTextEntry(Boolean): It will allows secureTextEntry when is sets true