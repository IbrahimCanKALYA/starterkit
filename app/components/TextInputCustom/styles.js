/* @flow */
import { Dimensions } from 'react-native';

const screenHeight = Dimensions.get('window').height;

const styles = {
  itemView: {
    marginTop: screenHeight / 35,
    justifyContent: 'center',
    borderWidth: 1,
    borderColor: '#D5D5D5',
    borderRadius: 5,
  },
  focusedItemView: {
    marginTop: screenHeight / 35,
    justifyContent: 'center',
    borderWidth: 1,
    borderColor: '#007aff',
    borderRadius: 5,
  },
  textInputStyle: {
    textAlign: 'left',
    fontSize: 17,
    fontFamily: 'Segoe UI',
    color: '#9A9A9A',
  },
};

export default styles;
