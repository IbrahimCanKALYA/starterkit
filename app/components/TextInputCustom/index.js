/* @flow */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, TextInput } from 'react-native';
import styles from './styles';

type Props = any;

type State = {
  text: string,
  focused: boolean,
};
class TextInputCustom extends Component<Props, State> {
  static defaultProps: any;

  constructor(props: Props) {
    super(props);

    this.state = {
      text: '',
      focused: false,
    };

    (this: any).onChangeText = this.onChangeText.bind(this);
    (this: any).getText = this.getText.bind(this);
  }

  onChangeText(text: string): void {
    const {
      props: { onChangeText },
    } = this;
    this.setState({ text });
    onChangeText(text);
  }

  getText(): string {
    const {
      state: { text },
    } = this;
    return text;
  }

  setText(text: string): void {
    this.setState({ text });
  }

  focus(): void {
    (this: any).textInput.focus();
  }

  clear(): void {
    const {
      props: { onChangeText },
    } = this;
    (this: any).text = '';
    onChangeText((this: any).text);
    (this: any).textInput.clear();
  }

  render() {
    const {
      state: { focused, text },
    } = this;
    const {
      props: { secureTextEntry, placeholder },
    } = this;
    return (
      <View style={focused ? styles.focusedItemView : styles.itemView}>
        <TextInput
          ref={(inputRef) => {
            (this: any).textInput = inputRef;
          }}
          value={text}
          onChangeText={this.onChangeText}
          onFocus={() => this.setState({ focused: true })}
          onBlur={() => this.setState({ focused: false })}
          style={styles.textInputStyle}
          autoCorrect={false}
          placeholder={placeholder}
          secureTextEntry={secureTextEntry}
        />
      </View>
    );
  }
}

TextInputCustom.propTypes = {
  onChangeText: PropTypes.func,
  placeholder: PropTypes.string,
  secureTextEntry: PropTypes.bool,
};
TextInputCustom.defaultProps = {
  onChangeText: () => {},
  placeholder: '',
  secureTextEntry: false,
};

export default TextInputCustom;
