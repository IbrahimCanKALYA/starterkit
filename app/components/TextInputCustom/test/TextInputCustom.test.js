import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TextInputCustom from '../index';

test('TextInputCustom renders correctly', () => {
  const tree = renderer
    .create(<TextInputCustom onChangeText={() => {}} placeholder="" secureTextEntry={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
