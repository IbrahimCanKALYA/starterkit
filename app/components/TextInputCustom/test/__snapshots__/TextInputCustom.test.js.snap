// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputCustom renders correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "#D5D5D5",
      "borderRadius": 5,
      "borderWidth": 1,
      "justifyContent": "center",
      "marginTop": 38.114285714285714,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder=""
    secureTextEntry={false}
    style={
      Object {
        "color": "#9A9A9A",
        "fontFamily": "Segoe UI",
        "fontSize": 17,
        "textAlign": "left",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
